reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -0,0 +1,262 @@
+		return new MonoProcessorSink<>(MonoProcessor.create());
+	}
+
+	// == interfaces ==
+
+	/**
+	 * A flavor of {@link SequenceSink} that is not attached to a single {@link Subscriber}
+	 * but rather viewable {@link #asFlux() as a Flux}. Most likely, such a {@link SequenceSink}
+	 * is capable of multicasting to several subscribers. However implementations can chose
+	 * to be unicast instead, and only accept one {@link Subscriber} at a time (as allowed
+	 * by the {@link org.reactivestreams.Publisher} specification).
+	 *
+	 * @param <T> the type of elements that can be emitted through this sink
+	 */
+	public interface StandaloneFluxSink<T> extends SequenceSink<T> {
+
+		@Override
+		StandaloneFluxSink<T> next(T t);
+
+		/**
+		 * Return the companion {@link Flux} instance that is backed by this sink.
+		 * All calls to this method return the same instance.
+		 *
+		 * @return the {@link Flux} view associated to this {@link StandaloneFluxSink}
+		 */
+		Flux<T> asFlux();
+
+	}
+
+	/**
+	 * A flavor of {@link ScalarSink} that is not attached to a single {@link Subscriber}
+	 * but rather viewable {@link #asMono() as a Mono}. Most likely, such a {@link ScalarSink}
+	 * is capable of multicasting to several subscribers. However implementations can chose
+	 * to be unicast instead, and only accept one {@link Subscriber} at a time (as allowed
+	 * by the {@link org.reactivestreams.Publisher} specification).
+	 *
+	 * @param <T> the type of elements that can be emitted through this sink
+	 */
+	public interface StandaloneMonoSink<T> extends ScalarSink<T> {
+
+		/**
+		 * Return the companion {@link Mono} instance that is backed by this sink.
+		 * All calls to this method return the same instance.
+		 *
+		 * @return the {@link Mono} view associated to this {@link StandaloneMonoSink}
+		 */
+		Mono<T> asMono();
+	}
+
+	// == concrete classes ==
+
+	static final class FluxProcessorSink<T>
+			implements StandaloneFluxSink<T> {
+
+		final FluxSink<T>         delegateSink;
+		final FluxProcessor<T, T> processor;
+
+		@SuppressWarnings("deprecation")
+		FluxProcessorSink(FluxProcessor<T, T> processor) {
+			this.processor = processor;
+			this.delegateSink = processor.sink();
+		}
+
+		@Override
+		public Flux<T> asFlux() {
+			return processor;
+		}
+
+		@Override
+		public void complete() {
+			delegateSink.complete();
+		}
+
+		@Override
+		public void error(Throwable e) {
+			delegateSink.error(e);
+		}
+
+		@Override
+		public StandaloneFluxSink<T> next(T t) {
+			delegateSink.next(t);
+			return this;
+		}
+	}
+
+	static final class MonoProcessorSink<T> implements StandaloneMonoSink<T> {
+
+		final MonoProcessor<T> processor;
+		boolean done;
+
+		MonoProcessorSink(MonoProcessor<T> processor) {
+			this.processor = processor;
+		}
+
+		@Override
+		public Mono<T> asMono() {
+			return this.processor;
+		}
+
+		@Override
+		public void success() {
+			synchronized (processor) {
+				if (done) {
+					return;
+				}
+				done = true;
```

Comment:
Just curious, why not using CAS?

Reply:
too much on my plate to have the perfect implementation in time for M1 :(